lib/components/UploadPicker.vue

```diff
@@ -294,16 +294,27 @@ export default defineComponent({
 		},
 
 		hasFailure(): boolean {
-			return this.queue?.filter((upload: Upload) => upload.status === UploadStatus.FAILED).length !== 0
+			return this.queue?.some((upload: Upload) => upload.status === UploadStatus.FAILED)
 		},
 		isUploading(): boolean {
 			return this.queue?.length > 0
 		},
-		isAssembling(): boolean {
-			return this.queue?.filter((upload: Upload) => upload.status === UploadStatus.ASSEMBLING).length !== 0
+		isOnlyAssembling(): boolean {
+			return !this.queue?.some((upload: Upload) => {
+				// ignore empty uploads or meta uploads
+				if (upload.size === 0) {
+					return false
+				}
+				// If all the uploads are assembling or finished, the ongoing task is assembling
+				return upload.status !== UploadStatus.ASSEMBLING && upload.status !== UploadStatus.FINISHED
+			})
```

Comment:
Negating here is harder to spot, instead we can make it easier by negate the condition:

```suggestion
			return this.queue.every((upload: Upload) => (
				// ignore empty uploads or meta uploads
				upload.size === 0
				// all the uploads are assembling
				|| upload.status === UploadStatus.ASSEMBLING
				// or finished
				|| upload.status === UploadStatus.FINISHED
			))
```